microprofile/WFLY-10712_microprofile_metrics.asciidoc

```diff
@@ -0,0 +1,230 @@
+= [WFLY-10712] Support Eclipse MicroProfile Metrics
+:author:            Jeff Mesnil
+:email:             jmesnil@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+:keywords:          observability,microprofile,metrics,openshift
+
+== Overview
+
+The goal of this feature is to increase observability of WildFly by exposing its metrics to monitoring tools such as https://prometheus.io[Promotheus]
+and the https://docs.openshift.com/container-platform/3.11/architecture/infrastructure_components/web_console.html[OpenShift Console].
+
+This feature adds support for the https://github.com/eclipse/microprofile-metrics/releases/download/1.1.1/metrics_spec-1.1.1.pdf[Eclipse MicroProfile Metrics 1.1.1 specification].
+
+The integration uses the https://github.com/smallrye/smallrye-metrics[smallrye-metrics] component to provide the MicroProfile Metrics implementation.
+
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-10712[WFLY-10712]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1026[EAP7-1026]
+* https://issues.jboss.org/browse/WFLY-10522[WFLY-10522] - Support Eclipse MicroProfile Config
+** MicroProfile Metrics has a dependency on MicroProfile Config
+* https://issues.jboss.org/browse/CLOUD-2731[CLOUD-2731] - Expose EAP Metrics so that it can be consumed by Kiali
+* https://issues.jboss.org/browse/EAP7-1060[EAP7-1060] - Make "Expose EAP Metrics" configurable in HAL Web Console
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:rsvoboda@redhat.com[Rostislav Svoboda]
+
+=== Affected Projects or Components
+
+The implementation of Eclipse MicroProfile Metrics 1.1.1 is provided by https://github.com/smallrye/smallrye-metrics[smallrye-metrics].
+
+=== Other Interested Projects
+
+* OpenShift Web Console MUST be able to consume metrics exposed by this extension.
+* Prometheus MUST be able to consume metrics exposed by this extension.
+* HAL, WildFly Web Console, will expose the new extension to configure it and potentially display the metrics exposed by the extension.
```

Comment:
Where is the RFE defined for HAL ?

Reply:
https://issues.jboss.org/browse/EAP7-1060 I'll add it to the issue metadata

Reply:
it's already in the proposal :)